### PR TITLE
Refactor functional api vectorized jacobian to use batched grad parameter

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #65566
* #65564
* #65565

This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

Differential Revision: [D31236257](https://our.internmc.facebook.com/intern/diff/D31236257)

### CODE REVIEW

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).

**Review by albanD — APPROVED** [review]
LGTM

File: torch/autograd/functional.py [hunk 1]
```diff
@@ -427,14 +427,10 @@ def jacobian(func, inputs, create_graph=False, strict=False, vectorize=False):
...
-            instead of once per row. This should lead to performance
-            improvements in many use cases, however, due to this feature
-            being incomplete, there may be performance cliffs. Please
-            use `torch._C._debug_only_display_vmap_fallback_warnings(True)`
-            to show any performance warnings and file us issues if
-            warnings exist for your use case. Defaults to ``False``.
+            ``True``, we perform only a single ``autograd.grad`` call with
+            ``batched_grad=True`` which uses the vmap prototype feature.
+            See :func:`torch.autograd.grad`'s ``batched_grad`` parameter for
+            more information.
```
Comment (zou3519) [inline torch/autograd/functional.py:433]: There should be a larger warning about the performance cliffs here

**Review by zou3519 — APPROVED** [review]

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).

**Pull request closed without merging**